executor/runtime/docker/docker.go

```diff
@@ -1497,23 +1504,54 @@ func (r *DockerRuntime) createOtherUserContainer(ctx context.Context, v1Containe
 	return containerCreateBody.ID, nil
 }
 
-func k8sContainerToDockerConfigs(v1Container v1.Container, mainContainerID string) (*container.Config, *container.HostConfig, *network.NetworkingConfig) {
+func (r *DockerRuntime) k8sContainerToDockerConfigs(v1Container v1.Container, mainContainerID string, mainContainerRoot string) (*container.Config, *container.HostConfig, *network.NetworkingConfig) {
 	// These labels are needed for titus-node-problem-detector
 	// to know that this container is actually part of the "main" one.
 	labels := map[string]string{
 		models.ExecutorPidLabel: fmt.Sprintf("%d", os.Getpid()),
 		models.TaskIDLabel:      mainContainerID,
 	}
+	mounts := []mount.Mount{
+		{
+			Type:     "bind",
+			Source:   mainContainerRoot + "/logs",
```

Comment:
Interesting - is this how Docker does it, under the covers?

Is the main container's `/logs` guaranteed to exist at this point?  Or I guess it doesn't matter because the main container will start first, and we'll create `/logs` anyway?

And just to confirm: if the user has `/logs` in their container pointing to `/path/to/somewhere/on/the/host`, we'll stomp over it, right?

Reply:
Haha no, docker 

> Interesting - is this how Docker does it, under the covers?
Haha no. Well sorta.
Normally volumes are their own separate thing under /var/lib/docker/volumes, and then *that* is bind mounted into the container.

Here I'm just picking a super specific host bind mount that *happens* to be the root dir of another container (the main one).

This is not something docker can normally support because it can't make assumptions about the storage driver.
We on titus can though, because we are using overlay. If we change that to use something else, then this trick will break.

Amazingly, this trick does work on docker-for-mac. I have no idea how though (closed source).

> 
> Is the main container's `/logs` guaranteed to exist at this point? Or I guess it doesn't matter because the main container will start first, and we'll create `/logs` anyway?
> 

Yes, I'm relying on the main container to be ready first, it gets /logs from the pushEnvironment function.


> And just to confirm: if the user has `/logs` in their container pointing to `/path/to/somewhere/on/the/host`, we'll stomp over it, right?

If it is a dir, tar won't care. If it is a file called 'logs' in '/', then tar will overwrite it because we use `AllowOverwriteDirWithFile: true` in our tar command.